File: Planning.API/Models/Classe.cs
using System;
using System.Collections.Generic;

namespace Planning.API.Models
{
    public class Classe
    {
        public int Id {get; set;}
        public string Nom {get; set;}
        public int NiveauId { get; set; }
        public int AnneeId { get; set; }

        public Niveau Niveau { get; set; }
        public Annee Annee { get; set; }

        public ICollection<ProfClasse> ClasseProfs { get; set;}
        public ICollection<CoursClasse> ClasseCours { get; set; }
        public ICollection<Eleve> Eleves { get; set; }
    }
}

File: Planning-SPA/src/app/view/classe/models/classe.ts
import { Prof } from './../../prof/models/prof';
import { Eleve } from '../../eleve/models/eleve';
export class Classe {
  id: number;
  niveauName: string;
  anneeName: string;
  professeurs: Prof[];
  eleves: Eleve[];

  constructor(
    public nom: string,
    public niveauId: number,
    public anneeId: number,
  ) {
    this.id = 0;
  }

}

